src/foam/classloader/WebModelFileDAO.js

```diff
@@ -53,9 +53,18 @@ foam.CLASS({
         var foamCLASS = foam.CLASS;
 
         foam.CLASS = function(m) {
-          var cls = m.class ? foam.lookup(m.class) : foam.core.Model;
-          model = cls.create(m, self);
-          foam.CLASS = foamCLASS;
+          var cls = m.class ? foam.lookup(m.class) :
+                foam.core.Model;
+          var mdl = cls.create(m, self);
+          if ( mdl.id === id ) {
```

Comment:
If the file name matches, but it contains multiple `CLASS` calls (some of which may not be of the requested id).